### ISSUE TITLE
Crash when validating a aws_iam_role_policy_attachment

### ISSUE THREAD

**Issue by robothor** [hashicorp/terraform#8185]
This is for Terraform v 0.7.0

### Affected Resource(s)
- aws_iam_role_policy_attachment

I wasn't sure if this resource supported a count parameter, so I tried the following:

### Terraform Configuration Files
```hcl
resource "aws_iam_role" "role" {
    name = "test-role"
}

resource "aws_iam_policy" "policy" {
    name = "test-policy"
    description = "A test policy"
    policy = <<EOF
{
  something
}
EOF
}

resource "aws_iam_policy" "policy2" {
    name = "test-policy"
    description = "A test policy"
    policy = <<EOF
{
  something
}
EOF
}

variable "foo" {
  default = ["${aws_iam_policy.policy.arn}", "${aws_iam_policy.policy2.arn}"]
}


resource "aws_iam_role_policy_attachment" "test-attach" {
  count = "${length(foo)}"
    role = "${aws_iam_role.role.name}"
    policy_arn = "${element(foo, count.index)}"
}
```

### Debug Output
Terraform output: https://gist.github.com/robothor/9e0aca5d1bc6c091026cdac5055e2c4d

### Panic Output
crash.log: https://gist.github.com/robothor/7dbdde22a7fe1e62f1747f22ba594ea0

### Expected Behavior
I would have expected a parse error or something similar, not a crash.

### Actual Behavior
Terraform crashed with the message that the issue should be reported.

### Steps to Reproduce
1. `terraform validate .`  (assuming an empty directory with main.tf)

**Comment by stack72** [comment]
Hi @robothor 

thanks for opening the issue here, sorry this has caused you some pain! So I am trying to recreate the conditions of the panic (so we can fix it!). The code snippet that you have posted behaves like this for me:

```
% terraform plan
Error configuring: 1 error(s) occurred:

* strconv.ParseInt: parsing "${length(foo)}": invalid syntax

% terraform apply                                                                                                                                                                                       1 ↵
Error configuring: 1 error(s) occurred:

* strconv.ParseInt: parsing "${length(foo)}": invalid syntax
```

Is the code above different from what caused you issues? 

thanks

paul

**Comment by mitchellh** [comment]
Got a repro down to a single resource (the resource type and name can be anything):

```
resource "aws_instance" "test-attach" {
  count = "${length(list)}"
}
```

**Comment by mitchellh** [comment]
Got a fix in the referenced PR #8243

**Comment by robothor** [comment]
Thanks!
